File: Desafios em JavaScript/1160 - Crescimento Populacional.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

function calculateYearsToExceed(populationA, populationB, growthRateA, growthRateB) {
  let years = 0;

  while (populationA <= populationB) {
    populationA += Math.floor(populationA * (growthRateA / 100));
    populationB += Math.floor(populationB * (growthRateB / 100));
    years++;

    if (years > 100) {
      return -1; // Mais de 1 século
    }
  }

  return years;
}

function printResult(years) {
  if (years === -1) {
    console.log("Mais de 1 seculo.");
  } else {
    console.log(years + " anos.");
  }
}

var numCases = parseInt(lines.shift());

for (let i = 0; i < numCases; i++) {
  var [populationA, populationB, growthRateA, growthRateB] = lines
    .shift()
    .split(" ")
    .map(Number);

  var years = calculateYearsToExceed(populationA, populationB, growthRateA, growthRateB);
  printResult(years);
}

File: Desafios em C#/1160 - Crescimento Populacional.cs
using System; 

//Developed by: @LucasMarcuzo

public class Beecrowd{
    
    public static void Main(string[] args){
        
        int numCases = Convert.ToInt32(Console.ReadLine());

        for (int i = 0; i < numCases; i++){
            
            string[] input = Console.ReadLine().Split(' ');
            int populationA = int.Parse(input[0]);
            int populationB = int.Parse(input[1]);
            double growthRateA = double.Parse(input[2]);
            double growthRateB = double.Parse(input[3]);

            int years = CalculateYearsToExceed(populationA, populationB, growthRateA, growthRateB);
            PrintResult(years);
        }
    }

    public static int CalculateYearsToExceed(int populationA, int populationB, double growthRateA, double growthRateB){
        
        int years = 0;

        while (populationA <= populationB){
            
            populationA += (int)(populationA * (growthRateA / 100));
            populationB += (int)(populationB * (growthRateB / 100));
            years++;

            if (years > 100)
            {
                return -1; // Mais de 1 século
            }
        }

        return years;
    }

    public static void PrintResult(int years){
        
        if (years == -1){
            
            Console.WriteLine("Mais de 1 seculo.");
        }
        else{
            
            Console.WriteLine(years + " anos.");
        }
    }
}
